guide.adoc

```diff
@@ -0,0 +1,596 @@
+
+It's worth pausing here a moment and realize that simply thanks to the addition of the FMP plugin to our Maven POM, our application is ready for the cloud. While it is important to understand the concepts at play here, it's also worth mentioning that simply running `./mvnw fabric8:deploy -Popenshift` while connected to a running OpenShift instance will deploy our app to the cloud and make it available without us having to worry (at least for now) about how it got there or what's required to do so! In fact, you can access your application by running
+
+```bash
+oc get route spring-boot-rest-http -o jsonpath='{"http://"}{.spec.host}{"\n"}'
+```
+
+and pasting that URL in your favorite browser.
+
+This works because FMP generated the `spring-boot-rest-http` `Route` automatically for you as follows:
+
+```yaml
+---
+apiVersion: v1
+kind: Route #<1>
+metadata:
+  labels:
+    app: spring-boot-rest-http
+    provider: fabric8
+    version: 1.5.14-1-SNAPSHOT
+    group: io.openshift.booster
+  name: spring-boot-rest-http #<2>
+spec:
+  port:
+    targetPort: 8080 #<3>
+  to:
+    kind: Service #<4>
+    name: spring-boot-rest-http #<5>
+```
+<1> Create a `Route`…
+<2> named `spring-boot-rest-http`…
+<3> targeting the `8080` port of …
+<4> the `Service`…
+<5> named `spring-boot-rest-http`!
+
+Let's now look at some the `spring-boot-rest-http-svc.yml` file, the `Service` definition:
+
+```yaml
+---
+apiVersion: v1
+kind: Service #<1>
+metadata:
+  annotations:  #<2>
+    fabric8.io/git-commit: 1fc0c37f21be5db9a77f93e42382583b32a588a7
+    fabric8.io/scm-con-url: scm:git:https://github.com/openshiftio/booster-parent.git/spring-boot-rest-http
+    prometheus.io/port: "9779"
+    fabric8.io/scm-url: https://github.com/openshiftio/spring-boot-rest-http
+    fabric8.io/iconUrl: img/icons/spring-boot.svg
+    fabric8.io/git-branch: guide
+    prometheus.io/scrape: "true"
+    fabric8.io/scm-devcon-url: scm:git:git:@github.com:openshiftio/booster-parent.git/spring-boot-rest-http
+    fabric8.io/scm-tag: booster-parent-23
+  labels:
+    expose: "true"  #<4>
+    app: spring-boot-rest-http #<6>
+    provider: fabric8
+    version: 1.5.14-1-SNAPSHOT
+    group: io.openshift.booster
+  name: spring-boot-rest-http #<3>
+spec:
+  ports:
+  - name: http #<5>
+    port: 8080
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    app: spring-boot-rest-http
+    provider: fabric8
+    group: io.openshift.booster
+```
+<1> Create a `Service`…
+<2> with some annotations, which provide metadata that interested third-parties can leverage…
+<3> named `spring-boot-rest-http`…
+<4> marked as being `expose`d, meaning a `Route` should be generated to expose the `Service` from outside the cluster…
```

Comment:
the "`" is misplaced in `exposed`

Reply:
Actually, that's on purpose since the keyword is `expose` not `exposed`, though maybe that doesn't render properly…

Reply:
Indeed, doesn't render properly. Fixing.